Automattic/jetpack
# AuthorRecommendation: add backend support

Followup PR for https://github.com/Automattic/jetpack/pull/30587

This PR adds jetpack endpoint for retrieving the user's followed sites.
In case the request is handled on wp com env then we use the library to retrieve this, otherwise, we send out a request to get this data.

Since users automatically follow blogs that they manage, this PR adds the option `filter_user_blogs` to filter the results returned by the backend. 

This PR changes the editor file to use the endpoint and work with real data. Also provides some minimalistic error handling.

Fixes 1721-gh-Automattic/dotcom-forge#issuecomment-1523158523

## Proposed changes:
- Add jetpack endpoint
- Connect endpoint with editor
- Wire `remove_user_blogs`

### Other information:


## Jetpack product discussion
pdDR7T-LK-p2
1721-gh-Automattic/dotcom-forge

## Does this pull request change what data or activity we track or use?
This block saves the subscribed sites that the user wants to include as block attributes. 
For each recommendation, its `blog_id`, `URL`, `name`, and `site_icon` is persisted.




## Testing instructions:






* Run `bin/jetpack-downloader test jetpack author-recommendation-backend` on your sandbox
* Add the author recommendations block to a page or to your site
* Ensure that your blog subscriptions are displayed properly
* Check which of the subscriptions you want to include/show
* Check out the rendered block on your site
